gh-54781: Move Lib/tkinter/test/test_ttk/ to Lib/test/test_ttk/

vstinner opened the pull request:
Move Lib/tkinter/test/ to Lib/test/test_tkinter/.

Rename test_tk to test_tkinter, and rename test_ttk_guionly to test_ttk.

<!-- gh-issue-number: gh-54781 -->
* gh-54781
<!-- /gh-issue-number -->

vstinner committed:
gh-54781: Move Lib/tkinter/test/test_ttk/ to Lib/test/test_ttk/

* Add Lib/test/test_ttk/__init__.py based on test_ttk_guionly.py.
* Remove Lib/test/test_ttk_guionly.py.

vstinner committed:
Move Lib/tkinter/test/ to Lib/test/test_tkinter/

vstinner committed:
Add Lib/test/test_tkinter/__init__.py

Remove Lib/test/test_tk.py.

vstinner committed:
Update imports

vstinner committed:
Update references

vstinner committed:
Add NEWS entry

serhiy-storchaka commented on Lib/test/test_tkinter/__init__.py:
```diff
@@ -12,9 +12,6 @@
-    return loader.discover('tkinter.test.test_tkinter')
-
 
-if __name__ == '__main__':
```
Does `python -m test.test_tkinter` still work?

merwok replied:
It would work if a `__main__` module gets added.  But it was `test.test_tk` before, not `test.test_tkinter`.

vstinner replied:
Ah you care about this command? If you do, I can add a ``__main__.py`` file.

serhiy-storchaka replied:
Yes, please add it. And test different ways of running tests:
```
python -m test.test_xxx
python -m unittest test.test_xxx
python -m test test_xxx
```

vstinner replied:
I added ``__main__`` sub-modules. Example:

```
./python -m test.test_ttk -v
./python -m test test_ttk -v -u all
./python -m unittest test.test_ttk -v
```

The 3 commands end with: ``Ran 308 tests``.

serhiy-storchaka commented on Lib/test/test_tkinter/__init__.py:
```diff
@@ -12,9 +12,6 @@
-if __name__ == '__main__':
-    unittest.main()
+def load_tests(*args):
+    return support.load_package_tests(os.path.dirname(__file__), *args)
```
Why not use just `loader.discover()`?

vstinner replied:
I copy/paste the code from test_asyncio, it works, so I didn't change it :-)

serhiy-storchaka replied:
Does `loader.discover()` work with zipimport?

vstinner replied:
> Does loader.discover() work with zipimport?

I have no idea. If there is an issue, I suggest to fix all load_tests() functions of Lib/test/ in a separated PR.

vstinner committed:
Add __main__ sub-modules

vstinner committed:
Add missing unittest import

Fix spelling typo: involving (missing G)

Remove useless "if __name__ == '__main__':" block from
Lib/test/test_ttk/__init__.py: the package has a __main__.py sub-module.

serhiy-storchaka reviewed (approved):
LGTM.

But I see that Git does not recognize renaming of test_tk.py. It breaks history and future backports. Can something be done with this?
